File: Algorithms/SierpinskyViewController.m
//
//  SierpinskyViewController.m
//  Algorithms
//
//  Created by Maxime Boulat on 2/7/17.
//  Copyright © 2017 Maxime Boulat. All rights reserved.
//

#import "SierpinskyViewController.h"

@interface SierpinskyViewController ()

@end

@implementation SierpinskyViewController

#pragma mark - Lifecycle

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
	
	
	[self addSubviewsToView:self.view];
	
}


- (void) addSubviewsToView: (UIView *)view {
	
	
	// base case
	
	if (view.frame.size.width < 1) {
		view.backgroundColor = [UIColor blueColor];
	}
	else {
		// first view
		
		CGRect rect = CGRectMake(0, 0, view.frame.size.width / 2, view.frame.size.height / 2);
		UIView * subview = [[UIView alloc]initWithFrame:rect];
		[view addSubview:subview];
		[self addSubviewsToView:subview];
		
		
		// second view
		
		rect = CGRectMake(view.frame.size.width / 2, 0, view.frame.size.width / 2, view.frame.size.height / 2);
		subview = [[UIView alloc]initWithFrame:rect];
		[view addSubview:subview];
		[self addSubviewsToView:subview];
		
		// third view
		
		rect = CGRectMake(view.frame.size.width / 2, view.frame.size.height / 2, view.frame.size.width / 2, view.frame.size.height / 2);
		subview = [[UIView alloc]initWithFrame:rect];
		[view addSubview:subview];
		[self addSubviewsToView:subview];
	}
}



@end

File: Algorithms/SierpinskyViewController.swift
//
//  SierpinskyViewController.swift
//  Algorithms
//
//  Created by Maxime Boulat on 2/7/17.
//  Copyright © 2017 Maxime Boulat. All rights reserved.
//

import UIKit

class SierpinskyViewController: UIViewController {
    
    // MARK: - Lifecycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        for subview in self.view.subviews {
            subview.removeFromSuperview()
        }
        
        addSubviews(to: self.view)
    }
    
    // MARK: - Helpers
    
    private func addSubviews(to view: UIView) {
        
        // base case
        if view.frame.size.width < 1 {
            view.backgroundColor = UIColor.blue
        } else {
            // first view
            var rect = CGRect(x: 0, y: 0, width: view.frame.size.width / 2, height: view.frame.size.height / 2)
            var subview = UIView(frame: rect)
            view.addSubview(subview)
            addSubviews(to: subview)
            
            // second view
            rect = CGRect(x: view.frame.size.width / 2, y: 0, width: view.frame.size.width / 2, height: view.frame.size.height / 2)
            subview = UIView(frame: rect)
            view.addSubview(subview)
            addSubviews(to: subview)
            
            // third view
            rect = CGRect(x: view.frame.size.width / 2, y: view.frame.size.height / 2, width: view.frame.size.width / 2, height: view.frame.size.height / 2)
            subview = UIView(frame: rect)
            view.addSubview(subview)
            addSubviews(to: subview)
        }
    }
} 